src/sqlancer/cnosdb/CnosDBGlobalState.java

```diff
@@ -0,0 +1,53 @@
+package sqlancer.cnosdb;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import sqlancer.ExecutionTimer;
+import sqlancer.GlobalState;
+import sqlancer.cnosdb.client.CnosDBConnection;
+import sqlancer.common.query.Query;
+
+public class CnosDBGlobalState extends GlobalState<CnosDBOptions, CnosDBSchema, CnosDBConnection> {
+
+    public static final char IMMUTABLE = 'i';
+    public static final char STABLE = 's';
+    public static final char VOLATILE = 'v';
+
+    private final Map<String, Character> functionsAndTypes = new HashMap<>();
+    private final List<Character> allowedFunctionTypes = Arrays.asList(IMMUTABLE, STABLE, VOLATILE);
+
+    @Override
+    protected void executeEpilogue(Query<?> q, boolean success, ExecutionTimer timer) throws Exception {
+        boolean logExecutionTime = getOptions().logExecutionTime();
```

Comment:
This seems to be a copy of https://github.com/sqlancer/sqlancer/blob/67751dc3d83b880cd8c7ed53416b8b157a2ba2ea/src/sqlancer/SQLGlobalState.java#L19-L27

I think it should be possible to inherit from `SQLGlobalState` instead?

Reply:
SQLGlobalState need SQLConnection. Currently our implementation does not use jdbc.

Reply:
Okay, I see. It would be ideal to still re-use some code (e.g., by factoring out methods), but this could also be as part of a follow-up PR.